Worker/src/Symphony/SessionPollingService.cs

```diff
@@ -149,18 +149,23 @@ private static TaskOptions InitializeDefaultTaskOptions()
 
     var taskRequests = payloads.Select(bytes =>
                                        {
-                                         var resultId = Guid.NewGuid()
-                                                            .ToString();
-                                         Logger.LogDebug("Create task {task}",
-                                                         resultId);
                                          return new TaskRequest
                                                 {
                                                   Payload = UnsafeByteOperations.UnsafeWrap(bytes.AsMemory(0,
                                                                                                            bytes.Length)),
 
                                                   ExpectedOutputKeys =
                                                   {
-                                                    resultId,
+                                                    TaskHandler.CreateResultsMetaDataAsync(new[]
```

Comment:
You should batch it before the select instead of making a request for each task.

Reply:
For now, I think we could keep the result creation one by one as in most cases, there is only few expected outputs.